modules/createHistory.js

```diff
@@ -69,21 +69,21 @@ function createHistory(options={}) {
   function getTransitionConfirmationMessage() {
     var message = null;
 
-    for (var i = 0, len = transitionHooks.length; i < len && typeof message !== 'string'; ++i)
+    for (var i = 0, len = transitionHooks.length; i < len && message === null; ++i)
```

Comment:
Using `message === null` here will break the loop after the very first function that returns `undefined`, which is any function that doesn't have a `return` statement. Let's use `message == null` instead.